src/WebJobs.Extensions.MobileApps/Config/MobileAppsConfiguration.cs

```diff
@@ -53,8 +62,194 @@ public void Initialize(ExtensionConfigContext context)
+            IValueBinder binder = (IValueBinder)Activator.CreateInstance(genericType, context);
+
+            return Task.FromResult(binder);
+        }
+
+        internal IMobileServiceClient BindForClient(MobileTableAttribute attribute)
+        {
+            MobileTableContext context = CreateContext(attribute);
+            return context.Client;
+        }
+
+        internal MobileTableContext BindForOutput(MobileTableAttribute attribute)
+        {
+            return CreateContext(attribute);
+        }
+
+        internal static Uri ResolveMobileAppUri(Uri defaultUri, string attributeUriString)
+        {
+            Uri resolvedUri = defaultUri;
+
+            if (!string.IsNullOrEmpty(attributeUriString))
+            {
+                string resolvedUriString = GetSettingFromConfigOrEnvironment(attributeUriString);
+                resolvedUri = new Uri(resolvedUriString);
+            }
+
+            return resolvedUri;
+        }
+
+        internal MobileTableContext CreateContext(MobileTableAttribute attribute)
+        {
+            Uri resolvedUri = ResolveMobileAppUri(MobileAppUri, attribute.MobileAppUri);
+            string resolvedApiKey = ResolveApiKey(ApiKey, attribute.ApiKey);
+
+            return new MobileTableContext
+            {
+                Client = GetClient(resolvedUri, resolvedApiKey),
+                ResolvedAttribute = attribute
+            };
+        }
+
+        internal static string ResolveApiKey(string defaultApiKey, string attributeApiKey)
+        {
+            string resolvedApiKey = defaultApiKey;
+
+            // If the attribute specifies an empty string ApiKey, set the ApiKey to null.
+            if (attributeApiKey == string.Empty)
+            {
+                resolvedApiKey = null;
+            }
+            else if (attributeApiKey != null)
+            {
+                resolvedApiKey = GetSettingFromConfigOrEnvironment(attributeApiKey);
+            }
+
+            return resolvedApiKey;
+        }
+
+        internal IMobileServiceClient GetClient(Uri mobileAppUri, string apiKey)
+        {
+            string key = GetCacheKey(mobileAppUri, apiKey);
+            return ClientCache.GetOrAdd(key, (c) => CreateMobileServiceClient(ClientFactory, mobileAppUri, apiKey));
+        }
+
+        internal static string GetCacheKey(Uri mobileAppUri, string apiKey)
+        {
+            return string.Format("{0};{1}", mobileAppUri, apiKey);
+        }
+
+        internal static IMobileServiceClient CreateMobileServiceClient(IMobileServiceClientFactory factory, Uri mobileAppUri, string apiKey = null)
+        {
+            HttpMessageHandler[] handlers = null;
+            if (!string.IsNullOrEmpty(apiKey))
+            {
+                handlers = new[] { new MobileServiceApiKeyHandler(apiKey) };
+            }
+
+            return factory.CreateClient(mobileAppUri, handlers);
         }
 
         internal static string GetSettingFromConfigOrEnvironment(string key)
```

Comment:
Can we avoid the direct dependency on ConfigurationManager / Environment and plumb thing through INameResolver?. 

Or we could extend Attribute Cloner with another form or [AutoResolve]   to handle this. But we really don't want extensions doing this adhoc. NotificationHub does it too; and does it slightly differently.